source/MaterialXCore/Definition.cpp

```diff
@@ -32,6 +33,31 @@ const string UnitDef::UNITTYPE_ATTRIBUTE = "unittype";
 // NodeDef methods
 //
 
+const string& NodeDef::getType() const
+{
+    const vector<OutputPtr>& activeOutputs = getActiveOutputs();
+
+    // Some organizational nodes have no output types
+    if (getNodeGroup() == ORGANIZATION_NODE_GROUP && activeOutputs.empty())
+    {
+        return NONE_TYPE_STRING;
+    }
+
+    size_t numActiveOutputs = activeOutputs.size();
+    if (numActiveOutputs > 1)
+    {
+        return MULTI_OUTPUT_TYPE_STRING;
+    }
+    else if (numActiveOutputs == 1)
+    {
+        return activeOutputs[0]->getType();
+    }
+    else
+    {
+        throw Exception("Nodedef: " + getName() + " has no outputs");
```

Comment:
I thought we changed this to return empty string and not throw an exception?

Reply:
This is the current code. If there is some other changes elsewhere then you'll need to merge this back to dev/....

Reply:
I would prefer to keep the "none" type and keep the exception here to report an error there is no  output. The API is much simpler and consistent if _all_ nodes have an output.

I think we could consider the "none" type to be similar to the "message" attribute type in Maya, used only for relationships between nodes. For example we could let the output on backdrop nodes connect to the nodes that are kept inside the backdrop. I'm not saying we should implement this now though.. My main concern is that by keeping "none" as a valid data type the API is simpler and more efficient.

Reply:
After talking, I've restored output as "none" for backdrop.